### ISSUE TITLE
Code block languages with spaces break code block detection

### ISSUE THREAD

**Issue by DBX12** [markdownlint/markdownlint#294]
## Description
Code blocks are not detected correctly if you use a language with spaces.

### Environment
`Linux devfreund13 5.0.0-37-generic #40~18.04.1-Ubuntu SMP Thu Nov 14 12:06:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

**MDL Version**
0.8.0

### Expected Behavior
Language names with spaces are detected correctly like other languages. Rules silencing feature works correctly because code block is detected correctly.

### Actual Behavior
Code blocks with names with a space are not detected as code block and thus other rules are activated when they are silenced (in my case MD013 got triggered despite this config entry `rule 'MD013', :code_blocks => false, :tables => false`; MD046 is left in default config)

## Replication Case

* Create `test.md` with this content (the code block fence inside the example is intentionally broken to keep github from interpreting them. Fix them on your machine)
* Use the config copied below.
* You will receive the error `test.md:4: MD013 Line length`
* Now replace the language `shell script` with `shell`
* Run mdl again and you see no error.

```markdown
# Hello world

` ``shell script
Ut doloremque veritatis suscipit laudantium laudantium suscipit. Ipsam deleniti sed fugiat rerum amet unde. Ut saepe aut nam. Quasi animi vitae eaque omnis maxime dolorum qui. Voluptatum omnis fugit sint id dolor maiores. Beatae iure ad aspernatur. Dolorem et beatae esse. Consectetur nostrum ut non.
` ``

```

Config
```
all
rule 'MD013', :code_blocks => false, :tables => false
```

**Comment by jaymzh** [comment]
To be nitpicky, "shell script" isn't a language, but it's an interesting point. I went looking at the CommonMark spec, to see what it says about this as I didn't think the language could be more than one word. And it turns out ... that while the "info string" can be anything that doesn't contain backticks (or tildas), the [*first word* is the language](https://spec.commonmark.org/0.29/#info-string) (though the spec isn't insistent on this):

```
An info string can be provided after the opening code fence. Although this spec doesn’t mandate any particular treatment of the info string, the first word is typically used to specify the language of the code block. In HTML output, the language is normally indicated by adding a class to the code element consisting of language- followed by the language name.
```

Since the `language-*` code tag element, afaict, never has spaces in it, I'd argue the language also cannot have spaces in it. In fact the CommonMark spec suggests all the other words be ignored (see [this example](https://spec.commonmark.org/0.29/#example-113)).

In the same sense that the spec says that trailing spaces after triple-backticks should be ignored, which we also error on.

So it's a bit of a gray area, but it seems this is probably the correct behavior. I'll leave this open for additional comment for a bit though.

Another option would be to have a `style` option for this rule that is something like `:ignore_multiword_info_strings` or somesuch.

**Comment by DBX12** [comment]
Personally, I prefer `bash` or similar as language name as well, but the Jetbrains Suite thinks otherwise. I did not find an option to change it. And if I use another language, I loose the syntax highlighting. :(

**Comment by jaymzh** [comment]
Interesting. OK, well, I think that's probably a pretty good reason to provide some sort of compatibility. It shouldn't be hard to at least create that option, as I mentioned. Seems like a small and reasonable feature request.

**Comment by NicolasMassart** [comment]
Also the code block first line could contain more than one word if you use extensions like [fenced-codeblock](https://python-markdown.github.io/extensions/fenced_code_blocks/):
```markdown
    ```bash tab="Syntax"
    # this is my comment in bash syntax about running ls
    ls -la
    ```
```
In this case, the code block is not considered as a block and the comment is considered as a first level title : `MD025 Multiple top level headers in the same document`.

Fixing this currently is impossible except by removing the `tab` part or the comment. None of these options are good IMO.

So having code block detected by a ` ``` ` followed by whatever would be the best for me.

**Comment by flugsio** [comment]
Another example, in this case the extra words are not used for specifying the language, but for extra parameters: https://github.com/dylanowen/mdbook-graphviz

```
    ```dot process
    digraph {
        processed -> graph
    }
    ```
```
